### PR TITLE
Add note about Warp syncing

### LINKED ISSUE #3887: [Feature Request] Document `--sync warp` as faster way of syncing
Warp syncing should be documented as faster way for syncing a pruned node.

### PR DESCRIPTION
**Pull request by DrW3RK**
Ref Issue - https://github.com/w3f/polkadot-wiki/issues/3887

### CODE REVIEW

File: docs/maintain/maintain-guides-how-to-validate-polkadot.md [hunk 1]
```diff
@@ -401,6 +401,15 @@ validator mode right away:
 
 :::
 
+:::tip Use Warp syncing for the fast syncing of a pruned node
+
+`./target/production/polkadot --sync warp`imports the latest blockchain state and proofs. While the
+warp sync is in progress, the `failed to associate send_message response to the sender` messages can
+be ignored.
```
Comment (bkchr) [inline docs/maintain/maintain-guides-how-to-validate-polkadot.md]: This message should not be shown anymore on latest master. 

And in general, these kind of things should be reported in issues and not added as "can be ignored" to the wiki.

File: docs/maintain/maintain-guides-how-to-validate-polkadot.md [hunk 2]
```diff
@@ -401,6 +401,15 @@ validator mode right away:
 
 :::
 
+:::tip Use Warp syncing for the fast syncing of a pruned node
+
+`./target/production/polkadot --sync warp`imports the latest blockchain state and proofs. While the
```
Comment (bkchr) [inline docs/maintain/maintain-guides-how-to-validate-polkadot.md]:
```suggestion
`./target/production/polkadot --sync warp` imports the latest blockchain state and proofs. While the
```

**Review by bkchr — COMMENTED** [review]
It is a little bit short, isn't it? Maybe some more words on how it works?

1. Warps to the latest finalized block by importing the headers of the blocks that enacted the new authorities. This happens every era.
2. After warping to the latest finalized block, the node is downloading the state of this block from its peers.
3. After state sync is finished, the block bodies from genesis will be downloaded. This happens in background, but the node is already ready to be "used" as validator/whatever.

Something like this, but better :P

File: docs/maintain/maintain-guides-how-to-validate-polkadot.md (hunk 1 above)
Comment (profradha) [reply docs/maintain/maintain-guides-how-to-validate-polkadot.md]: Agreed. Removed it.

**Comment by DrW3RK** [comment]
@bkchr Added more context and content, as requested.

File: docs/maintain/maintain-guides-how-to-validate-polkadot.md [hunk 3]
```diff
@@ -403,7 +403,14 @@ validator mode right away:
 
 :::tip Use Warp syncing for the fast syncing of a pruned node
 
-`./target/production/polkadot --sync warp` imports the latest blockchain state and proofs.
+`./target/production/polkadot --sync warp`
+
+Warp sync initially downloads the whole state of the latest finalized block, and the finality proofs
```
Comment (bkchr) [inline docs/maintain/maintain-guides-how-to-validate-polkadot.md]: It first downloads the finality proofs and validates them and then downloads the state of the latest finalized block.

Comment (profradha) [reply docs/maintain/maintain-guides-how-to-validate-polkadot.md]: Edited it. Probably need to suggest this change to the accepted answer on Substrate StackExchange as well, as it is ambiguous https://substrate.stackexchange.com/questions/334/what-kinds-of-sync-mechanisms-does-substrate-implement

**Review by bkchr — APPROVED** [review]

**Pull request merged**